### DISCUSSION TITLE
Swift like language update?

### DISCUSSION THREAD

**Discussion by metacritical** [cappuccino/cappuccino#3090] [Ideas]
Any Plans to do a Swift like language implementation for this?

**Comment by enquora** [comment]
We do not plan to implement a Swift-like language in Cappuccino. Objective-J, our current language, is designed to be very simple and can be learned within 1 or 2 days. In contrast, Swift is a complex language with a less regular syntax, and its benefits are most significant for systems programming rather than user application development. Note also that a very large amount of native Objective-C code is available publicly - this can be converted to Objective-J very easily. The same is not at all true for Swift and a (hypothetical) version for Cappuccino.

The strengths of Cocoa, which Cappuccino emulates, are closely tied to the way Objective-C and Objective-J are implemented. This implementation ensures the robustness and ease of use that developers rely on.

For those new to Objective-J or Cappuccino, we offer assistance in our Gitter channel and encourage joining and asking questions. Syntax highlighting is available for most programmers' text editors, including Visual Studio Code, Vim, Textmate-derivatives and Xcode.
